python/tk_houdini/base_hooks/node_handler_base.py

```diff
@@ -0,0 +1,471 @@
+        """
+        Set up the node handler class to contain the appropriate settings from the
+        shotgun configs.
+
+        We do this here as it only needs to happen on the first instantiation of
+        a class type. When contexts change the modules are reloaded and this is
+        wiped. 
+
+        Prevents us needing to define, in code, shotgun templates and settings.
+        """
+        if cls.NODE_TYPE:
+            if hasattr(cls, "_work_template"):
+                return super(NodeHandlerBase, cls).__new__(cls, *args, **kwargs)
+            engine = sgtk.platform.current_engine()
+            node_handlers = engine.get_setting("node_handlers")
+            for handler in node_handlers:
+                if handler["node_type"] == cls.NODE_TYPE:
+                    work_template = engine.get_template_by_name(handler["work_template"])
+                    setattr(cls, "_work_template", work_template)
+                    publish_template = engine.get_template_by_name(handler["publish_template"])
+                    setattr(cls, "_publish_template", publish_template)
+                    setattr(cls, "extra_args", handler["extra_args"])
+        return super(NodeHandlerBase, cls).__new__(cls, *args, **kwargs)
+
+    @staticmethod
+    def generate_callback_script_str(method_name):
+        """
+        Helper function to generate a callback script string for houdini parameters.
+
+        :param str method_name: The name of the method we want to be our callback.
+
+        :rtype: str
+        """
+        callback_str = (
+            "__import__('sgtk').platform.current_engine().node_handler(kwargs['node'])"
+            ".{method}(kwargs)"
+        ).format(method=method_name)
+        return callback_str
+
+    def get_work_template(self, node):
+        """
+        Get the shotgun work template for this node handler.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :rtype: An :class:`sgtk.Template` instance.
+        """
+        return self._work_template
+
+    def get_publish_template(self, node):
+        """
+        Get the shotgun publish template for this node handler.
+
+        :param node: A :class:`hou.Node` instance.
+
+        :rtype: An :class:`sgtk.Template` instance.
+        """
+        return self._publish_template
+        
+    def _get_template(self, template_name):
+        """
+        Get a shotgun template from the given name.
+
+        :param str template_name: The name of the template to get.
+
+        :rtype: An :class:`sgtk.Template` instance.
+
+        :raises: :class:`sgtk.TankError` if no template name supplied
+            or the template doesn't exist.
+        """
+        template_name = self.extra_args.get(template_name)
+        if not template_name:
+            raise sgtk.TankError("No template name defined")
+        template = self.parent.get_template_by_name(template_name)
+        if not template:
+            raise sgtk.TankError("Can't find template")
```

Comment:
Make the messages more meaningful for the current context:
```suggestion
            raise sgtk.TankError(
                "No template name '{}' defined for node type '{}'"
                "".format(template_name, self.NODE_TYPE)
            )
        template = self.parent.get_template_by_name(template_name)
        if not template:
            raise sgtk.TankError(
                "Can't find template called '{}' defined for node type '{}'"
                "".format(template_name, self.NODE_TYPE)
            )
```